ui/src/table.tsx

```diff
@@ -692,29 +797,39 @@ export const
           } else {
             filters[filterKey] = filters[filterKey].filter(f => f !== filterVal)
           }
-
-          filter(filters)
-          search()
-          setGroups(groups => {
-            if (groups) initGroups()
-            return groups
-          })
+          if (m.pagination && m.events?.includes('filter')) {
+            wave.emit(m.name, 'filter', filters)
+            setCurrentPage(1)
+          } else {
+            filter(filters)
+            search()
+            setGroups(groups => {
```

Comment:
Is user able to use custom groups with custom filter implementation if `setGroups` is inside `else` part of a condition?

Reply:
Yes, groups have to be provided server side. This piece of code is for client-side, which does not matter in this context.